Cubical/Algebra/CommRing/Localisation/InvertingElements.agda

```diff
@@ -160,29 +160,21 @@ module InvertingElementsBase (R' : CommRing ℓ) where
       p : x' ≡ x
       p = eq/ _ _ ((1r , ∣ 0 , refl ∣₁) , path)
        where
-       useSolver1 : ∀ a b c → 1r · (a · b) · c ≡ a · (b · c)
-       useSolver1 = solve R'
-       useSolver2 : ∀ a b → a · b ≡ 1r · a · b
-       useSolver2 = solve R'
        path : 1r · (r · f ^ (l ∸ n)) · f ^ n ≡ 1r · r · f ^ l
-       path = 1r · (r · f ^ (l ∸ n)) · f ^ n ≡⟨ useSolver1 _ _ _ ⟩
+       path = 1r · (r · f ^ (l ∸ n)) · f ^ n ≡⟨ solve! R' ⟩
```

Comment:
So the solver can parse `f ^ n` as a single variable so to speak? This is great!

Reply:
Yes, it should be able to detect anything which is a variable applied to arguments, definition applied to some arguments or constructor applied to some arguments. If it encounters something outside scope, it will throw an error.

Reply:
I was inspired by code in the 1lab